api/IDBRequest.json

```diff
@@ -182,6 +182,50 @@
               "standard_track": true,
               "deprecated": false
             }
+          },
+          "improved_reporting_large_value_read_failures": {
+            "__compat": {
+              "description": "Improved reporting for large value read failures",
```

Comment:
@caugner I'm pretty sure this is the one: https://github.com/w3c/IndexedDB/pull/430.

The exceptions are listed in the spec at https://www.w3.org/TR/IndexedDB/#exceptions.

Reply:
Based on https://github.com/w3c/IndexedDB/issues/423, I would suggest to rename the feature as follows:

```suggestion
          "throws_NotReadableError": {
            "__compat": {
              "description": "Throws `NotReadableError` on storage read failures",
```

My impression is that "large value" is Chrome terminology, not part of the spec.

/cc @Elchi3 for a second opinion, as we may see more of these very specific behavioral features in the future.

Reply:
PS: We could add this spec_url: `https://www.w3.org/TR/IndexedDB/#:~:text=NotReadableError`

Reply:
I like @caugner's description and ID a lot more but I also find it strange to use both, sub features and array support statements here. Is this the summary?


```
"chrome": [
  {
   "version_added": "132",
   "notes": "Throws `NotReadableError` and `UnknownError`"
  }, 
 {
   "version_added": "130",
   "notes": "Throws `NotFoundError` and `DataError`"
  }, 
 {
   "version_added": "48",
   "notes": "Throws `DOMExceptions`"
  },
  {
   "version_added": "23",
   "notes": "Throws `DOMError`"
  }
],
```

Reply:
Thanks for the comments folks. This has been a bit of a weird tricky one to add to BCD, so I'm glad for the input.

In my latest commit, I have:

- Renamed it to `read_errors` — I have removed the Chrome-specific terminology, but [in the corresponding content updates](https://github.com/mdn/content/pull/37041#discussion_r1870427801) we agreed that generally these errors relate to problems with reading the data value from disk, and "read error" was the general terminology we settled on.
  - Note that this is not just about Chrome behavior, but rather Chrome updating its behavior to align with Firefox and then browsers updating to align with specified behavior. 
  - Also note that this is not just to do with `NotReadableError` exception type support, but also other related behavior changes.
- Added the spec URL as suggested. I'm pleased to see that you can use text fragments in spec URLs and that Firefox now supports text fragments ;-)
- removed all mention of throwing. This is a key point that came up in the content review — these exceptions are not thrown, rather they are returned when you query the `IDBRequest.error` property.

I don't really feel like @Elchi3's summary quite works — the change from using `DOMError`s to using `DOMException`s is a high-level change to reported error structures, where the thing I am adding is more to do with specific error types.

Reply:
I've updated the feature name as suggested.

I agree it would be nice to include other browser support information, but I've got no idea how to find that out. I can't find a mention of `NotReadableError` in the Gecko source code so, although they do support the same kind of error handling for read errors, I'm not sure if it is using the same exception names.

From the [spec bug](https://github.com/w3c/IndexedDB/pull/430), I am wondering whether this has bene agreed as the error to put in the spec, which matches Fx behavior, but Fx hasn't actually implemented the official specified error yet. In which case, could we call it something like "Specified reporting for read errors", and it would be accurate to say that only Chrome has implemented it so far?

Reply:
I'm not 100% sure, but most likely it was Firefox 14 that implemented `NotReadableError` in https://bugzilla.mozilla.org/show_bug.cgi?id=730161, cf. https://github.com/mozilla/gecko-dev/commit/1ce6cdf75fa16b85022e394f56e14ab2f691e7e8#diff-38a23a8920ed3a6e05068b536b2ba2f20b2ef6449a7a084ee0bc76faeca8e6d5R45

Reply:
Webkit most likely implemented it in https://bugs.webkit.org/show_bug.cgi?id=102137  / https://github.com/WebKit/WebKit/commit/b3684910d150ffa50b489a8f2b50d768f863bec2, i.e. Safari 537.43.1 before Safari 7 (537.71).

Reply:
That sounds good enough for me ;-)